### PR TITLE
MaximumSubArraySum

### PR DESCRIPTION
**Pull request by Ayse-Sadioglu**
### **Describe your change:**

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### **Checklist:**

- [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [ ] This pull request is all my own work -- I have not plagiarized.
- [ ] I know that pull requests will not be merged if they fail the automated tests.
- [ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [ ] All new Java files are placed inside an existing directory.
- [ ] All filenames are in all uppercase characters with no spaces or dashes.
- [ ] All functions and variable names follow Java naming conventions.
- [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: src/main/java/com/thealgorithms/others/MaximumSubArraySum.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            return Arr[ firstindex ];//it means array contains only one element
+            //and this element equals maximum subarray sum.
+        }
+
+        // Finding middle element's index of the array
+        int middle = ( firstindex + lastindex ) / 2;
+
+        // Finding maximum subarray sum for the left subarray
+
+        int leftMax = -1000000000;
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/MaximumSubArraySum.java:19]: Please use  Integer.MIN_VALUE

File: src/main/java/com/thealgorithms/others/MaximumSubArraySum.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        int sum = 0;
+        for (int i = middle; i >= firstindex; i--) {
+            sum += Arr[ i ];//addition for sum
+            if (sum > leftMax) {
+                leftMax = sum;
+            }
+        }
+
+        // Find maximum subarray sum for the right subarray
+        int rightMax = -1000000000;//should be minimum value
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/MaximumSubArraySum.java:29]: And here

File: src/main/java/com/thealgorithms/others/MaximumSubArraySum.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
+package others;
+
+public class MaximumSubArraySum {
+    //Maximum Subarray Sum using Divide and Conquer algorithm
+
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/MaximumSubArraySum.java:5]:
```suggestion

```

File: src/main/java/com/thealgorithms/others/MaximumSubArraySum.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        // Recursive part
+        int maxLeftRight = Math.max (MaximumSum (Arr, firstindex, middle),//left part of the array
+                MaximumSum (Arr, middle + 1, lastindex));//right part of the array
+
+        // return maximum of the three
+        return Math.max (maxLeftRight, leftMax + rightMax);
+    }
+
+
+}
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/MaximumSubArraySum.java:47]:
```suggestion
    }
}

```

**Review by siriak — CHANGES_REQUESTED** [review]
Looks good, please add some tests as in https://github.com/TheAlgorithms/Java/pull/2871/files

**Comment by siddhant2002** [comment]
Can I work under this issue?

**Comment by siriak** [comment]
@siddhant2002 yes, feel free to implement this

**Review by 22aam — COMMENTED** [review]
the code looks good to me

**Comment by siddhant2002** [comment]
@siriak please check the pr

**Comment by siriak** [comment]
@siddhant2002 which one? I've already reviewed this one

**Comment by siddhant2002** [comment]
@siriak please check the updated pr

**Comment by siriak** [comment]
Superseded by #2913

**Pull request closed without merging**